crud_exec_test.go

```diff
@@ -262,6 +264,45 @@ func (me *crudExecTest) TestScanStructs() {
 	assert.Equal(t, noTags[1].Name, "Test2")
 }
 
+
+func (me *crudExecTest) TestColumnRename() {
+	t := me.T()
+	// different key names are used each time to circumvent the caching that happens
+	// it seems like a solid assumption that when people use this feature,
+	// they would simply set a renaming function once at startup,
+	// and not change between requests like this
+	lowerAnon := struct {
+		FirstLower string
+		LastLower string
+	}{}
+	lowerColumnMap, lowerErr := getColumnMap(&lowerAnon)
+	assert.NoError(t, lowerErr)
+
+	var lowerKeys []string
+	for key, _ := range lowerColumnMap {
+		lowerKeys = append(lowerKeys, key)
+	}
+	assert.Contains(t, lowerKeys, "firstlower")
+	assert.Contains(t, lowerKeys, "lastlower")
+
+	// changing rename function
+	SetColumnRenameFunction(strings.ToUpper)
```

Comment:
I believe the tests are failing because of this line. I think this exposes what maybe a larger concern which is that this is changing the functionality at a global level. Im starting to think this should be provided as an argument and something that the dataset maintains so you could have different instances of datasets each with potentially different transform methods, that also help prevent tests from inadvertently leak into other tests.

What do you think?

Reply:
That's actually what I hoped to do initially, to have this whole thing be more "object-oriented", but since the `createColumnMap` method is pretty divorced from a dataset or database, and it seems like it would take a bit of finagling to pass a relevant object down through all the layers. But we could probably figure it out :smile: